feat: New Clear Search Button

popkinj opened the pull request:
- Button beside the sidebar title that clears both the search results and the text in the search title.
- Hides when not searching anything

![Screenshot from 2025-01-19 10-51-18](https://github.com/user-attachments/assets/a866cf7b-3c9b-4368-b370-e4585bb491dc)
![Screenshot from 2025-01-19 10-51-10](https://github.com/user-attachments/assets/03bd2522-6263-4c37-b581-35e0b7744540)


---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-epd-organics-info-355-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-epd-organics-info-355-frontend.apps.silver.devops.gov.bc.ca/api/)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-epd-organics-info/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-epd-organics-info/actions/workflows/merge.yml)

popkinj commented:
![Screenshot from 2025-01-20 12-13-32](https://github.com/user-attachments/assets/e790c493-ba10-45e6-8a9a-2bb003348502)
![Screenshot from 2025-01-20 12-13-40](https://github.com/user-attachments/assets/3d518768-386c-419d-a2ea-64b2d71c4b14)

kdbelair commented:
@popkinj Looks good - just noticing that the number of filters is being cut off at the top of screen on mobile.
![statuscutoff](https://github.com/user-attachments/assets/578d5573-98cd-4c87-a287-106c26da2dd1)

dawnbcgov commented:
There are a couple of issues I'm seeing:
- the buttons now are a different size than the search bar 
- It looks like the borders are 2px when they should be 1px
- there is a white border now when active, which we don't need
- it looks like the search tray names haven't been updated
- I don't think the zoom on hover feels particularly clean
- I think carrying over the notification to the other buttons is interesting, but I wonder why we stopped using the notification colour from the mockups? Also there are clipping issues on mobile as seen in Konane's screenshot
 
On PR: 
![image](https://github.com/user-attachments/assets/4cef4160-1c17-4d8b-a51a-963a70ca2829)
![image](https://github.com/user-attachments/assets/51a56f1e-9294-4d85-94a5-abcde65c9804)

On mockups:
![image](https://github.com/user-attachments/assets/56f6306b-43a5-4847-860f-f74f13f9818b)
![image](https://github.com/user-attachments/assets/27a6ed15-a588-4ce2-9157-c4f3e5ccf4a3)

popkinj commented:
Thanks for the feedback @dawnbcgov and @kdbelair . I'll take a look.

popkinj commented:
Latest updates:
- Replace chip with red dot
- Updated text in bottom draw to match the buttons (Search & Filter)
- Removed the underline in the autosearch dropdown.
- Removed the hover effect for the filter/search buttons

![red_dot_desktop](https://github.com/user-attachments/assets/73560f1d-e244-4369-838f-8f1fa292ae28)
![red-dot-mobile](https://github.com/user-attachments/assets/523dbdf6-3c23-40d0-b912-cb661960da35)

kdbelair commented:
The latest changes you have flagged in your latest update look good and are working well on my end. I understand you might be addressing these additional notes/observations in a different PR, but just flagging them here to keep track:

- the buttons are a different size than the search bar
- It looks like the borders are 2px when they should be 1px
- there is a white border now when active, which we don't need

I also noted this, but this would also change on the text search page too:
- the filter name of "Land Application Biosolids" should be updated to "Land Application" as shown in the Figma

popkinj commented:
> The latest changes you have flagged in your latest update look good and are working well on my end. I understand you might be addressing these additional notes/observations in a different PR, but just flagging them here to keep track:
> 
>     * the buttons are a different size than the search bar
> 
>     * It looks like the borders are 2px when they should be 1px
> 
>     * there is a white border now when active, which we don't need
> 
> 
> I also noted this, but this would also change on the text search page too:
> 
>     * the filter name of "Land Application Biosolids" should be updated to "Land Application" as shown in the Figma

Correct @kdbelair . I would like to merge my existing PRs before I address the button issues. :)

kdbelair commented:
Thanks @popkinj  - approved.

popkinj commented:
Working on these fixes now.

popkinj commented:
I could probably tweak these styles all week long hahaha. 
Think things are looking good now.
![Screenshot from 2025-01-27 11-20-59](https://github.com/user-attachments/assets/ea7e4442-9ad5-406f-83ce-c5da84d4d4da)
![Screenshot from 2025-01-27 11-25-59](https://github.com/user-attachments/assets/9b1eb924-0e28-4163-aa79-aea407579fe4)

kdbelair commented:
@popkinj -The status, filter and find me buttons still look slightly thicker than the search bar on desktop. Can you take a quick look if it can be adjusted? Everything else looks great

kdbelair commented:
Confirming that the size of the icons has been fixed. Good to go from my end.
